index.php

```diff
@@ -28,51 +28,23 @@ function error_page($header, $body)
     error_page('Configuration Error', 'Endpoint not yet configured, visit <a href="setup.php">setup.php</a> for instructions on how to set it up.');
 }
 
-//temp code generation to protect against CSRF, this is only going to be valid for between 2 and 4 minutes
-function generate_csrf_code($redirect_uri, $client_id, $state)
+function create_signed_code($key, $message, $ttl = 31536000, $appended_data = '')
```

Comment:
random pet-peeve, can you just add a comment that 31536000 is 365 days, the question of what a random number is exactly always bothers me when i read code at a later date.